common/runtime/model.go

```diff
@@ -534,3 +534,14 @@ func (a AdditionalWorkerNodePool) ValidateHAZonesUnchanged(currentAdditionalWork
 	}
 	return nil
 }
+
+func (a AdditionalWorkerNodePool) ValidateMachineTypesUnchanged(currentAdditionalWorkerNodePools []AdditionalWorkerNodePool) error {
```

Comment:
I wonder if it would be feasible to separate duties here.

1. Predicate checking if any machine type has been changed (possibly with positive logic)
2. Action - returning error